### PR TITLE
fix[javalib]:Runtime#availableProcessors now respects cpuset on small Linux systems

### LINKED ISSUE #3963: Available runtime processors is not really number of available processors
https://github.com/scala-native/scala-native/blob/8b1359c3f030433e6ce2c2b42a062f5c6e0808f6/javalib/src/main/scala/java/lang/Runtime.scala#L109

This function returns the number of available processors by reading from `_SC_NPROCESSORS_ONLN`, which utilizes `get_nprocs()`. The `get_nprocs()` function operates similarly to `lscpu` and does not indicate the number of processors a program can access.

Here is a code to show the difference:
```cpp
#include <stdio.h>
#include <stdlib.h>
#include <sys/sysinfo.h>
#include <omp.h>

int main()
{
        printf("This system has %d processors configured and "
                        "%d processors available.\n",
                        get_nprocs_conf(), get_nprocs());

        printf("and in OMP: %d\n", omp_get_max_threads());
        exit(EXIT_SUCCESS);
}
```

The output for different numbers of available threads (using `taskset`) is as follows:
```terminal
➜ g++ a.cpp -fopenmp
➜ taskset -c 0-0 ./a.out
This system has 8 processors configured and 8 processors available.
and in OMP: 1
➜ taskset -c 0-1 ./a.out
This system has 8 processors configured and 8 processors available.
and in OMP: 2
➜ taskset -c 0-4 ./a.out
This system has 8 processors configured and 8 processors available.
and in OMP: 5
➜ taskset -c 0-100 ./a.out
This system has 8 processors configured and 8 processors available.
and in OMP: 8

```

This may cause ambiguity in what *available* processors mean.

### PR DESCRIPTION
**Pull request by LeeTibbert**
Weakly fixes #3963 

`Runtime#availableProcessors` on small Linux systems now reports the number of logical processors allow
 by the process `cpuset`. 

This number may be smaller than that reported by `sysconf(_SC_NPROCESSORS_ONLN)` due to various
Linux mechanisms: cgroups, taskset, cpuset, etc.

'small' is defined as less than or equal to, currently, 1024. Above that limit `sysconf(_SC_NPROCESSORS_ONLN)` is
reported, same as before this PR.

See Issue for discussion and manual testing instructions (next sprint). Most notably, OpenMP variables are not considered.

### CODE REVIEW

File: javalib/src/main/resources/scala-native/sys/linux_sched.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /* On error, return -1. Let caller handle errors.
+     *
+     * On success, return the number of logical processors available to
+     * the process. Due to taskset, cpuset, and such, this may be
+     * less than sysconf number of online logical processes
+     * (_SC_NPROCESSORS_ONLN) which may be less than the total number of
+     * logical processes configured for the system (_SC_NPROCESSORS).
+     *
+     * Note that OpenMP environment variables are neither consulted nor used.
+     * This consistent with Java.
```
Comment (natsukagami) [inline javalib/src/main/resources/scala-native/sys/linux_sched.c:24]:
```suggestion
     * This is consistent with Java.
```

**Review by natsukagami — APPROVED** [review]
Looks great, thank you so much! @amsen20 please check if it does the right thing :)

**Comment by amsen20** [comment]
It works for me, thanks :)

**Review by amsen20 — APPROVED** [review]
I checked it locally and it worked

**Comment by LeeTibbert** [comment]
natsukagami, amsen20   

Thank you both for your review & suggestion/correction.  

It is nice to know that the proposed solution actually works in the use case and not just on my machines.

If either of you think there is a short term need for either > 1024 or for FreeBSD, please create an Issue.
I do not really have a NetBSD machine available, so someone else would have to champion that.

Somehow, I think we will all be back here again.

**Comment by LeeTibbert** [comment]
There is one failing Windows Test. I appears to be an intermittent failure, 
and unrelated to the substance of this PR.

Otherwise all Green & ready for review.

**Review by WojciechMazur — APPROVED** [review]

**Pull request merged**